### DISCUSSION TITLE
Best file system for Kobo SD card?

### DISCUSSION THREAD

**Discussion by jonnyl2** [koreader/koreader#11218] [Q&A] [answered]
I was wondering if any performance enhancement can be achieved by switching to a certain file system?

For Windows, MacOS, and Linux compatibility reasons I have it currently formatted with FAT32, but would be willing to give up on some compatibility for better performance. (I've read that UDF is also compatible with all 3 OS, but not sure if that'd be a better choice.)

Thanks all.

**Comment by poire-z** [comment, marked as answer]
> I was wondering if any performance enhancement can be achieved by switching to a certain file system?

I think not/none.
KOReader doesn't do many or big reads & writes, so the filesystem type shouldn't matter.
Except after a reboot at startup when it needs to read all the .lua files - but after boot and the first startup, all these files are probably cached in RAM and won't be reread from the filesystem if you restart KOReader.
As for book files themselves (and the CRE cache in cr3cache/), the filesystem read time is probably peanuts compared to the CPU work with the data read that is happening at the same time and after.

What kind of performace you think you are missing?

**Comment by jonnyl2** [comment]
Thanks for your answer.

>What kind of performace you think you are missing?

Browsing folder trees with many files is noticeably slower on the SD card compared to the on-board storage. It could just be a slow card, of course. But I wanted to make sure I make the most of it.

**Reply by poire-z** [reply]
I don't remember what is your device. With onboard + sdcard ?
On my Kobo:
`/dev/mmcblk0p3 on /mnt/onboard type vfat (rw,noatime,nodiratime,fmask=0022,dmask=0022,codepage=cp437,iocharset=iso8859-1,shortname=mixed,utf8,errors=remount-ro)
`
So the /mnt/onboard/ partition (from the internal multipartitionned sdcard) is also VFAT / FAT32.
So, it's probably the quality of your external SDCard vs. the internal memory/SDCard.

**Reply by jonnyl2** [reply]
It's a Kobo Aura H2O (1st edition)

> So, it's probably the quality of your external SDCard vs. the internal memory/SDCard.

Makes sense. Thank you!

**Comment by NiLuJe** [comment]
Even for the external slot, I'm unsure how well the native system would take to a different filesystem (without patching some udev scripts, at least).

So, yeah, it sucks, but I also think it's unlikely to change much (unless you have pathological directory structures) anyway.

**Reply by NiLuJe** [reply]
And, yeah, on Kobo, as @poire-z just showed, even on devices that run off an internal SD card (as opposed to an eMMC), that's also using FAT32 (and that takes even *more* patching to change ^^).

**Reply by NiLuJe** [reply]
As a fun counterpoint, the latest batch of Kindle Scribe firmwares have disabled USBMS and switched to MTP, and that was accompanied by switching the affected partition from FAT32 to ext4 ;).

**Reply by jonnyl2** [reply]
Good to know. Thank you!
